eng/tools/typespec-validation/src/tsv-runner-host.ts

```diff
@@ -34,8 +34,18 @@ export class TsvRunnerHost implements TsvHost {
     return readFileImpl(path, "utf-8");
   }
 
-  runCmd(cmd: string, cwd: string): Promise<[Error | null, string, string]> {
-    return runCmd(cmd, cwd);
+  runFile(file: string, args: string[], cwd?: string): Promise<[Error | null, string, string]> {
+    return runFile(file, args, cwd);
+  }
+
+  runNpm(args: string[], cwd?: string): Promise<[Error | null, string, string]> {
+    const [file, defaultArgs] = process.platform === "win32" ?
+      // Only way I could find to run "npm" on Windows, without using the shell (e.g. "cmd /c npm ...")
+      // "C:\Program Files\nodejs\node.exe", ["C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js"]
+      [process.execPath, [join(dirname(process.execPath), 'node_modules', 'npm', 'bin', 'npm-cli.js')]] :
```

Comment:
This is the scariest change in this PR, but as long as unit and functional tests pass on both windows and linux, it should be trustable.

Reply:
Since this avoids involving `cmd.exe` I believe this prevents shell injection by invoking node.exe directly. Maybe a `--` before the `npm-cli.js` path would further clarify that node arguments have ended and to treat the next argument as a script file to execute. 


https://nodejs.org/api/cli.html#--

> Indicate the end of node options. Pass the rest of the arguments to the script. If no script filename or eval/print script is supplied prior to this, then the next argument is used as a script filename.